### PR TITLE
create all shift dump

### PR DESCRIPTION
**Pull request by apackin**
#### What's this PR do?
Creates a CSV of all the shifts within the designated 50 day range
#### How should this be manually tested?
run: node console shiftsDump go
#### What are the relevant tickets?
https://admin.crisistextline.org/jira/secure/RapidBoard.jspa?rapidView=5&view=detail&selectedIssue=INT-180

### CODE REVIEW

**Comment by tongxiang** [comment]
For the record: 


Tong Xiang [4:49 PM]  
Checking out #160 now. How did you fix the CSV escaping issue earlier?

[4:49]  
Why did you decide not to include a CSV converting library?

assaf [4:51 PM]  
The escape issue was fixed by running csvsql with a designated ‘,’ delimeter. 
I decided not to use a library because we are renaming some of the keys, and I thought it would be easier to do it manually than figuring out how to modify the libraries

File: config.js [hunk 1]
```diff
@@ -126,7 +126,10 @@ CONFIG.time_interval = {
   days_until_older_user: 21,
 
   // Number of days in a week
-  one_week: 7
+  one_week: 7,
+
+  // Number of days in the future to dump shifts for
+  dumpShiftsDays: 28,
 };
```
Comment (jrauschenberg) [inline config.js]: I thought this was for a 50-day range. I'm sure you have a reason for this 28-day window, just want to be sure.

File: tasks/shiftsDump.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      Object.keys(shifts[0]).slice(5).join(',');
+    let CSVFormattedString = CSVKeys + '\n';
+    let values;
+    let value;
+
+    shifts.forEach((shift) => {
+      values = [];
+      for (let key in shift) {
+        if (shift.hasOwnProperty(key)) { 
+          value = '' + shift[key] + '';
```
Comment (jrauschenberg) [inline tasks/shiftsDump.js]: Can you just use '' + shift[key] here?

File: test/helpers/checkHTTPResponses.js [hunk 3]
```diff
@@ -53,7 +52,8 @@ function sendRequest (term, params) {
   //Get all time off requests within timeOffSearchParams
   WhenIWork.get(term, params, function(response) {
     console.log(`${term} returns ${response}`);
-    fs.writeFile(`sampleResponse/httpResponse1${capitalizeFirstLetter(term)}.json`, JSON.stringify(response), (err) => {
+    fs.writeFile(`sampleResponse/httpResponse1${capitalizeFirstLetter(term)}.json`, 
+                 JSON.stringify(response), (err) => {
       if (err) throw err;
```
Comment (jrauschenberg) [inline test/helpers/checkHTTPResponses.js]: Where is this thrown error being picked up?

**Comment by jrauschenberg** [comment]
Three minor comments, otherwise good to go!

**Comment by tongxiang** [comment]
from discussion yesterday, changes: 
1) adding more requests so we pull shifts either a) from Jan 2016 or b) pulling shifts a designated time range in the past 
2) adding some sort of sanity-checking / monitoring. maybe with stathat

File: test/helpers/checkHTTPResponses.js [hunk 4]
```diff
@@ -53,8 +52,9 @@ function sendRequest (term, params) {
   //Get all time off requests within timeOffSearchParams
   WhenIWork.get(term, params, function(response) {
     console.log(`${term} returns ${response}`);
-    fs.writeFile(`sampleResponse/httpResponse1${capitalizeFirstLetter(term)}.json`, JSON.stringify(response), (err) => {
-      if (err) throw err;
+    fs.writeFile(`sampleResponse/httpResponse1${capitalizeFirstLetter(term)}.json`, 
+                 JSON.stringify(response), (err) => {
+      if (err) console.log(err);
       console.log('It\'s saved!');
```
Comment (jrauschenberg) [inline test/helpers/checkHTTPResponses.js]: CONSOLE_WITH_TIME here?

**Comment by jrauschenberg** [comment]
Just one comment. Nice code!

File: tasks/shiftsDump.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    CONSOLE_WITH_TIME(err);
+  });
+};
+
+function getShifts () {
+  const options = {};
+  const promiseArray = [];
+
+  const today = new Date();
+  const firstDayOfYear = moment(new Date(today.getFullYear(), 1, 1));
```
Comment (tongxiang) [inline tasks/shiftsDump.js]: 1. `firstDayOfYear` is returning Feb 1st, 2016 for me. Is that desired behavior?
2. What are the second and third params you're passing into `moment()` doing?
3. What happens when it becomes 2017? Will this pull neglect any 2016 shifts?

File: tasks/shiftsDump.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  let totalDaysBack = firstDayOfYear.diff(moment(), 'days'); 
+
+  // When we request too many days at a time WiW times out
+  // To fix we are breaking it up into seperate request for every 30 days
+  while (totalDaysBack+30<CONFIG.time_interval.dumpShiftsDays) {
+      options.start = `${totalDaysBack} days`;
+      totalDaysBack+=30;
+      options.end = totalDaysBack<0 ?
+        `${totalDaysBack} days`:
+        options.end = `+${totalDaysBack} days`;
```
Comment (tongxiang) [inline tasks/shiftsDump.js]: is this a valid ternary operator? to have another assignment statement of `options.end` in the third section?

Comment (apackin) [inline tasks/shiftsDump.js]: Nope. Good catch.

**Comment by tongxiang** [comment]
👍  good to merge!

**Pull request merged**